src/main/webapp/app/exam/participate/summary/exam-result-summary.component.ts

```diff
@@ -253,5 +323,115 @@ export class ExamResultSummaryComponent implements OnInit {
         return false;
     }
 
+    isAfterResultsArePublished() {
+        if (this.isTestRun || this.isTestExam) {
+            return true;
+        }
+        if (this.studentExam?.exam?.publishResultsDate) {
+            return this.serverDateService.now().isAfter(this.studentExam.exam.publishResultsDate);
+        }
+        return false;
+    }
+
+    private getExerciseInfos(studentExamWithGrade?: StudentExamWithGradeDTO): Record<number, ResultSummaryExerciseInfo> {
+        const exerciseInfos: Record<number, ResultSummaryExerciseInfo> = {};
+        for (const exercise of this.studentExam?.exercises ?? []) {
+            if (exercise.id === undefined) {
+                console.error('Exercise id is undefined', exercise);
```

Comment:
Please don't just log the exception, but send it to sentry for later inspection (see course rounding)